### ISSUE TITLE
iOS Build Error: Include of a non-modular header inside framework module 'BugsnagReactNative.Bugsnag'

### ISSUE THREAD

**Issue by fbartho** [bugsnag/bugsnag-react-native#466]
### Description
We have code that is importing `<BugsnagReactNative/BugsnagReactNative.h>` from this, we were calling `[BugsnagReactNative start];` inside of a section of code (called from `appDidFinishLaunching`)

### Issue

![Screenshot of error](https://user-images.githubusercontent.com/209712/86410788-4c426480-bc70-11ea-9861-abe45020f888.png)

This wasn't an issue, until we upgraded one of our cocoapods dependencies (ZendeskChat) to their V2.

I'm a maintainer of `react-native-zendesk-chat` and I'm trying to upgrade it.

<details>
<summary>That updated pod spec has these lines:</summary>

```ruby
Pod::Spec.new do |s|
  s.name         = "RNZendeskChat"
  s.version      = package['version']
  s.summary      = package['description']
  s.license      = package['license']
  s.authors      = package['author']
  s.homepage     = package['homepage']
  s.platform     = :ios, "10"
  s.source       = { :git => "https://github.com/taskrabbit/react-native-zendesk-chat.git", :tag => "v#{s.version}" }
  s.source_files = "ios/*.{h,m}"
  s.static_framework = true

  s.framework    = 'Foundation'
  s.framework    = 'UIKit'

  s.dependency 'React'
  s.dependency 'ZendeskChatSDK'
end
```
</details>

### Environment

Library versions:

```shell
$ yarn list --pattern "(react-native|bugsnag)" # we don't have code-push
yarn list v1.22.4
├─ @expo/react-native-action-sheet@3.6.0
├─ @react-native-community/cli-debugger-ui@4.9.0
├─ @react-native-community/cli-platform-android@4.9.0
├─ @react-native-community/cli-platform-ios@4.9.0
├─ @react-native-community/cli-server-api@4.9.0
├─ @react-native-community/cli-tools@4.9.0
├─ @react-native-community/cli-types@4.9.0
├─ @react-native-community/cli@4.9.0
├─ @react-native-community/masked-view@0.1.9
├─ @react-native-community/netinfo@5.6.2
├─ @react-native-community/slider@2.0.9
├─ @stream-io/react-native-simple-markdown@1.1.2
├─ @types/react-native-fbsdk@1.0.0
├─ @types/react-native-permissions@2.0.0
├─ @types/react-native@0.62.12
├─ bugsnag-react-native@2.23.6
├─ css-to-react-native@2.3.2
├─ eslint-plugin-react-native-globals@0.1.2
├─ eslint-plugin-react-native@3.8.1
├─ jest-react-native@18.0.0
├─ metro-react-native-babel-preset@0.58.0
├─ metro-react-native-babel-transformer@0.58.0
├─ modal-react-native-web@0.2.0
├─ react-native-actionsheet@2.4.2
├─ react-native-config@0.12.0
├─ react-native-country-picker-modal@1.10.0
├─ react-native-document-picker@3.2.4
├─ react-native-fbsdk@1.1.2
├─ react-native-flipper@0.34.0
├─ react-native-gesture-handler@1.6.1
├─ react-native-image-crop-picker@0.28.0
├─ react-native-image-pan-zoom@2.1.11
├─ react-native-image-picker@0.28.1
├─ react-native-image-zoom-viewer@2.2.27
├─ react-native-iphone-x-helper@1.2.1
├─ react-native-keychain@4.0.5
├─ react-native-kochava-tracker@1.3.1
├─ react-native-permissions@2.0.10
├─ react-native-rate@1.2.1
├─ react-native-reanimated@1.8.0
├─ react-native-safe-area-context@0.7.3
├─ react-native-safe-area-view@0.14.9
├─ react-native-schemes-manager@2.0.0
├─ react-native-screens@2.4.0
├─ react-native-tab-view@2.13.0
├─ react-native-typescript-transformer@1.2.13
├─ react-native-webview@7.6.0
├─ react-native-zendesk-chat@0.3.1
├─ react-native@0.62.2
├─ reactotron-react-native@5.0.0
├─ stream-chat-react-native-core@0.10.2
│  └─ metro-react-native-babel-preset@0.53.1
├─ stream-chat-react-native@0.10.2
└─ urbanairship-react-native@6.0.1
```

- cocoapods version (if any) (`pod -v`): `1.9.3` (Note that the new command for this is `pod --version`)
- iOS/Android version(s): 12.2
- simulator/emulator or physical device?: Simulator
- debug mode or production?: Debug

- [x] (iOS only) `[BugsnagReactNative start]` is present in the
  `application:didFinishLaunchingWithOptions:` method in your `AppDelegate`
  class?

### Interesting Podfile details

```ruby
platform :ios, '11.0'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

# …snipped for brevity

	use_native_modules!
	use_frameworks!
```
Also, we have flipper disabled!

### Example code snippet

Code is just:
```objc
import <BugsnagReactNative/BugsnagReactNative.h>

// in appDidFinishLaunching:
[BugsnagReactNative start];
```

<details><summary>Error messages:</summary>

```


Showing All Messages
/Users/fbarthelemy/Code/TRClient5/node_modules/bugsnag-react-native/cocoa/vendor/bugsnag-cocoa/Source/Bugsnag.h:30:9: Include of non-modular header inside framework module 'BugsnagReactNative.Bugsnag': '../node_modules/bugsnag-react-native/cocoa/vendor/bugsnag-cocoa/Source/BugsnagPlugin.h'

/Users/fbarthelemy/Code/TRClient5/ios/TRClient/AppDelegate.m:23:9: Could not build module 'BugsnagReactNative'
```

Error is in Bugsnag.h:
![Screenshot of the error in Bugsnag.h](https://user-images.githubusercontent.com/209712/86411817-83197a00-bc72-11ea-94fd-5b28afa7faf3.png)

</details>

Potentially related ticket: #442

**Comment by fbartho** [comment]
*Workaround* is to allow non-modular headers at a Project level (temporarily -- as this is considered a problem in one of your libraries -- and probably a sign that other build weirdness will happen)

**Comment by steve-nester-uk** [comment]
Thanks for raising @fbartho, I agree that the temporary solution is not ideal. We are going to look into a better fix by moving the import into the .m file (tbc).

**Comment by steve-nester-uk** [comment]
@fbartho, I've just learned that a patch was applied in [2.23.7](https://github.com/bugsnag/bugsnag-react-native/releases/tag/v2.23.7) to the missing BugsnagPlugin.h header to the podspec file.  Please can you upgrade to 2.23.7 and confirm if this fixes the issue?

**Comment by fbartho** [comment]
@steve-nester-uk -- I tested against 2.23.9, and there's a new error:

![image](https://user-images.githubusercontent.com/209712/87109017-68b64200-c218-11ea-893d-67aec4fa6d87.png)

Trying 2.23.7 now.

**Comment by fbartho** [comment]
Problem is resolved with exactly `2.23.7` -- Do you need me to file a different Github Issue about `2.23.9` or is this sufficient?

**Comment by xander-jones** [comment]
Hey @fbartho, thanks for letting us know. We're already aware of this issue (moving to `v2.23.8` or `v2.23.9`) you can track it here: https://github.com/bugsnag/bugsnag-react-native/issues/462
